File: packages/compiler/ir-types/fern-ir-typescript/src/model/services/http/HttpEndpoint.ts
import { WithDocs } from "../../commons/WithDocs";
import { HttpAuth } from "./HttpAuth";
import { HttpHeader } from "./HttpHeader";
import { HttpMethod } from "./HttpMethod";
import { HttpRequest } from "./HttpRequest";
import { HttpResponse } from "./HttpResponse";
import { PathParameter } from "./PathParameter";
import { QueryParameter } from "./QueryParameter";

export interface HttpEndpoint extends WithDocs {
    endpointId: string;
    path: string;
    method: HttpMethod;
    headers: HttpHeader[];
    pathParameters: PathParameter[];
    queryParameters: QueryParameter[];
    request: HttpRequest;
    response: HttpResponse;
    auth: HttpAuth;
}

File: packages/compiler/ir-types/fern-ir-java/ir-api/model/src/main/java/com/fern/types/services/http/HttpEndpoint.java
package com.fern.types.services.http;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fern.StagedBuilderStyle;
import com.fern.interfaces.IWithDocs;
import java.lang.String;
import java.util.List;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableHttpEndpoint.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface HttpEndpoint extends IWithDocs {
  String endpointId();

  String path();

  HttpMethod method();

  List<HttpHeader> headers();

  List<PathParameter> pathParameters();

  List<QueryParameter> queryParameters();

  HttpRequest request();

  HttpResponse response();

  HttpAuth auth();

  static ImmutableHttpEndpoint.EndpointIdBuildStage builder() {
    return ImmutableHttpEndpoint.builder();
  }
}
